### ISSUE TITLE
would it be possible to change the area type of individual navmesh polygons?

### ISSUE THREAD

**Issue by bm212** [Unity-Technologies/NavMeshComponents#6]
I'm not sure if this is the right place to raise this as it's not directly connected to this particular change.

It would be great if I could select a navmesh polygon and change its area type without needing to rebake the navmesh.

Use case - I have 2 characters, one shoots at a target.  The other wants to move without straying into the field of fire of the first character. It seems that one way to do this would be to increase the cost of all navmesh polygons traversed by the field of fire (by changing them to an expensive area type), thus making the pathfinder tend to avoid them.

Apologies if this isn't an appropriate place to raise this request.

**Comment by jakobhu** [comment]
This is something we'd like to support in the future - one drawback with simply changing the area type of the existing polygons - is that the tessellation might not match the actual affected area. 
The NavMeshModifierVolume can be used with runtime updating the navmesh to create a similar effect - where the volume bounds control the tessellation in addition to the area type.

**Comment by IntegersOfK** [comment]
I'm trying to create a path where my AI people will preferentially choose to take areas they've already travelled. I came to this library and saw "bake navmesh at runtime" and this nice concept of NavMesh modifying. I've tried combining the game's traditional navmesh with the agent from this library to get a kind of "hybrid" situation, but I still can't find a way to update the cost of an "area" at runtime.

My other friend explained to me that he is making a lake, dam, and stream. The stream will get bigger so he'd like to change the landscape, and therefore the navmesh. This library is perfect, except again it's hard to explain to the system that the water should be passable but at a different cost than the beach.
